File: easy/Excel-Sheet-Column-Number-171.java
class Solution {
public int titleToNumber(String s) {
    if(s==null || s.length() == 0){
        throw new IllegalArgumentException("Input is not valid!");
    }
 
    HashMap<Character, Integer> map = new HashMap<Character, Integer>();
    char c = 'A';
    for(int i=1; i<=26; i++){
        map.put(c, i);
        c += 1; 
    }
 
    int result = 0;
    int i = s.length()-1;
    int t = 0;
    while(i >= 0){
        char curr = s.charAt(i);
        result = result + (int) Math.pow(26, t) * map.get(curr);
        t++;
        i--;
    }
 
    return result;
}
}

File: easy/Excel-Sheet-Column-Number-171.cpp
class Solution {
public:
    int titleToNumber(string s) {
        int n = s.size();
        int res = 0;
        unsigned int tmp = 1;
        for (int i = n-1; i >= 0; --i) {
            res += (s[i] - 'A' + 1) * tmp; 
            tmp *= 26;
        }
        return res;
    }
};
